fdbclient/include/fdbclient/MetaclusterManagement.actor.h

```diff
@@ -1583,33 +1605,29 @@ Future<std::vector<std::pair<TenantName, TenantMapEntry>>> listTenants(
 
 	loop {
 		try {
-			tr->setOption(FDBTransactionOptions::READ_SYSTEM_KEYS);
-			tr->setOption(FDBTransactionOptions::READ_LOCK_AWARE);
 			if (filters.empty()) {
-				wait(store(results, listTenantsTransaction(tr, begin, end, limit + offset)));
+				wait(store(results, MetaclusterAPI::listTenantsTransaction(tr, begin, end, limit + offset)));
```

Comment:
Since we are only using the names returned from this query, I think it would be ideal if we avoided doing the two-step listing procedure unless we need to for the filtering. The return type of the function used by fdbcli could be the one that returns the name ID pairs.